### DISCUSSION TITLE
Feature Request: Highlight Code Mixed with Markdown Text in a Different Color

### DISCUSSION THREAD

**Discussion by ghost** [imfing/hextra#40] [Ideas]
Many websites highlight code blocks when used with regular text. Examples:
1. Wikipedia - the "History" section [here](https://en.wikipedia.org/wiki/%22Hello,_World!%22_program) has different color for code "hello, world", "main", "printf", etc.

2. Exercise - the "Instructions" section [here](https://exercism.org/tracks/go/exercises/weather-forecast/).

Highlighted code mixed with text makes it easier to read.

**Comment by imfing** [comment]
I understand. 
I would recommend use [Custom CSS](https://imfing.github.io/hextra/docs/advanced/customization/#custom-css) to add [background color](https://developer.mozilla.org/en-US/docs/Web/CSS/background-color) to the [`<code>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/code) element.

**Reply by ghost** [reply]
This changes the background color of code block in 

```
print("Hello, World!")
```

and, not the text in `text`

**Reply by imfing** [reply]
I think this would work:

<img width="1459" alt="image" src="https://github.com/imfing/hextra/assets/5097752/f3bdde7a-3bc6-4030-9b9a-41208b82bf8c">

Code `assets/css/custom.css`:
```css
.content code:not(.code-block code) {
  background-color: "replace-with-your-color";
}
```

**Reply by ghost** [reply]
My `assets/css/custom.css` file has:

```
.content code:not(.code-block code) {
  background-color: #c97c2e;
}
```

and I did

```
hugo mod get -u github.com/imfing/hextra@main
hugo server --buildDrafts --disableFastRender --ignoreCache
```

but don't see any changes :(

**Reply by imfing** [reply]
It should work, I just test it in the starter template on Gitpod:

<img width="1066" alt="image" src="https://github.com/imfing/hextra/assets/5097752/fdf77dac-538b-4c36-bb59-20eeb9649c79">

Hugo version:

```
$ hugo version
hugo v0.117.0-b2f0696cad918fb61420a6aff173eb36662b406e+extended linux/amd64 BuildDate=2023-08-07T12:49:48Z VendorInfo=brew
```

How did you structure your project?

**Reply by ghost** [reply]
I'm using this theme as a Hugo module and I'm on version `0.118`

This is the repository:
https://codeberg.org/mishra/katalog.ci

[This](http://localhost:1313/docs/stack/valid-parenthesis/) is an example page which contains code blocks and code formatted text.

**Reply by imfing** [reply]
It works for me locally, I didn't make any changes

```
hugo server -D -F --disableFastRender
```

<img width="1289" alt="image" src="https://github.com/imfing/hextra/assets/5097752/43ca2006-ee8a-4ed9-ade4-8668cde45c0c">

**Reply by ghost** [reply]
This works only in light mode :)

In the left menu bar if you change Light to Dark it doesn't work.

I restarted my browser, cleared data, etc.

Is there a way to make it work in dark mode ?

**Reply by imfing** [reply]
Oh, I forgot the dark mode. You can change it by:

```css
:is(html[class~=dark] .content code:not(.code-block code)) {
    background-color: red;
}
```

<img width="1015" alt="image" src="https://github.com/imfing/hextra/assets/5097752/a81e139a-7727-47e1-8643-811e71754734">


Next time, you can find the CSS rule by opening Chrome developer tool, then select the element, and find the CSS rule for it:

![image](https://github.com/imfing/hextra/assets/5097752/1014948b-b1db-4fae-8c53-dcdf7c7e30eb)

**Reply by ghost** [reply]
The color should work for both modes. Using the below CSS does the trick:

```
.content code:not(.code-block code) {
  color: #c97c2e;
}
```

Changing `background-color` to `color` works for both light and dark mode.

I can create a PR for documenting this on [this page](https://imfing.github.io/hextra/docs/advanced/customization/)

**Reply by imfing** [reply]
No need for a PR, I think people can this thread via GitHub search.

**Reply by ghost** [reply]
I think people may find it more intuitive to read documentation on Hextra's website than search GitHub discussions. 

I think this should be on the Hextra website to showcase customization. 

I've created PR #55 for documenting this discussion and discussion #46.

**Comment by ghost** [comment]
Replacing `background-color` with `color` in the CSS file does it !!

It also works in both modes.
